storage/src/adb/mod.rs

```diff
@@ -43,3 +46,36 @@ pub enum Error {
     #[error("key not found")]
     KeyNotFound,
 }
+
+/// Utility to align the sizes of an MMR and location journal pair, used by keyless, immutable &
+/// variable adb recovery. Returns the aligned size.
+async fn align_mmr_and_locations<E: Storage + Clock + Metrics, H: Hasher>(
+    mmr: &mut crate::mmr::journaled::Mmr<E, H>,
```

Comment:
nit: not sure we need to be so verbose with the argument types?

Reply:
fixed, was getting a bit sloppy with these.